### ISSUE TITLE
Can't do any other client operations inside subscribe block

### ISSUE THREAD

**Issue by sinecode** [cloudamqp/amqp-client.rb#18]
It's not possible to call any high level operations inside the subscribe block. The program halts and it never goes on.

Example (publish inside subscribe)

```ruby
client = AMQP::Client.new(RABBITMQ_URL).start
my_queue = client.queue('myqueue')
my_other_queue = client.queue('output.queue')
my_queue.subscribe(prefetch: 20, worker_threads: 0, no_ack: false) do |msg|
  # the client will hang here
  my_other_queue.publish('done')
  msg.ack
  puts 'acked'
rescue StandardError => e
  warn(e)
  msg.reject(requeue: false)
end
```

**Comment by carlhoerberg** [comment]
The problem is `worker_threads: 0`, without it it works... add a `sleep` after the `do` block instead

**Comment by carlhoerberg** [comment]
forcing worker_threads to be > 0 in the high level client. You can replcate the same behavior as your example with: 

```ruby
tag, threads = my_queue.subscribe(prefetch: 20, worker_threads: 1, no_ack: false) do |msg|
  # the client will hang here
  my_other_queue.publish('done')
  msg.ack
  puts 'acked'
rescue StandardError => e
  warn(e)
  msg.reject(requeue: false)
end
threads.each(&:join)
```

**Comment by sinecode** [comment]
Hi @carlhoerberg. Thanks for taking your time into this.

I know that this is an issue when `worker_threads: 0` only, but I think that this is actually a bug that should be fixed. I encountered this bug while I was working with this gem and I had very hard time to debug my program that randomly halted even if I didn't nothing wrong on my side. At least there should be a warning on the documentation signaling that no high level operations are allowed when `worker_threads: 0`.

[In my PR](https://github.com/cloudamqp/amqp-client.rb/pull/19) this bug is fixed. I know that the test I wrote was wrong: I posted a comment with a new test that fails without the path (it hangs forever).
